src/bp/ui-studio/src/web/components/Layout/StatusBar/BotSwitcher.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { Fragment } from 'react'
+import { Dropdown, Glyphicon } from 'react-bootstrap'
+import style from './StatusBar.styl'
+import ActionItem from './ActionItem'
+import _ from 'lodash'
+
+export default class BotSwitcher extends React.Component {
+  constructor(props) {
+    super(props)
+  }
+
+  state = {
+    open: false
+  }
+
+  toggleOpen = () => this.setState({ open: !this.state.open })
+
+  render() {
+    return (
+      <Fragment>
+        <ActionItem
+          id="statusbar_switchbot"
+          title="Switch Bot"
+          description="Switch to an other bot. This will leave this interface."
+          onClick={this.toggleOpen}
+        >
+          <Glyphicon glyph="twitter" style={{ marginRight: '5px' }} />
+          <span>{this.props.currentBotId}</span>
+        </ActionItem>
+
+        <Dropdown pullRight dropup={true} open={this.state.open} onToggle={this.toggleOpen} id="bot-switcher">
+          {/* react-bootstrap warning otherwise */}
+          <Dropdown.Toggle style={{ display: 'none' }} />
+          <Dropdown.Menu pullRight onClose={this.toggleOpen} className={style.langSwitherMenu}>
+            {this.props.botIds
+              .filter(id => this.props.currentBotId != id)
+              .map(id => (
+                <li className={style.langItem}>
+                  <a key={id} href={`/studio/${id}`}>
+                    {id}
+                  </a>
+                </li>
+              ))}
+          </Dropdown.Menu>
+        </Dropdown>
+      </Fragment>
```

Comment:
Can you use blueprint instead for that? There is an example popover & menu in the admin UI (UserDropdownMenu). I was wondering if it would be a good idea to add double-click to open the admin ui

Also replace the glyphicon by Blueprint's Icon

Reply:
👍 I based my work on the lang switcher, I might refactor both at the sametime.

Reply:
Yeah, i'm really not a fan of the lang menueither :p I've added the flag and didn't touch it, because of all the required logic to programatically handle the popover, keyboard events to switch between languages, etc